### PR TITLE
D8CORE-1472: Updates to lockup settings for logo support

### PR DESCRIPTION
**Pull request by boznik**
# READY FOR REVIEW

# Summary
- Adding lockup options and the ability to upload a custom logo
- Allows lockup to have only a logo
- Updates lockup form to hide/show fields that are available on each option.

# Review By (Date)
- EOD Friday the 24th if you can.

# Urgency
- Medium / High

# Steps to Test

1. Do a fresh build of the 1.x branch of the acsf-cardinalsites stack with the stanford_profile
2. Check out this branch
3. Clear caches
4. Go to /admin/appearance/settings/stanford_basic
5. Select different lockup options from the drop down and see the available line input fields hide/show corresponding to the option selected.
6. Select an option other than `Logo Only` and fill out the line fields.
7. Save form and review on the home page you see what you would have expected to see.
8. Repeat with another option other than logo only.
9. Select `Logo Only` and save the form.
10. Review the home page and only see the Stanford wordmark.
11. Go back to the /admin/appearance/settings/stanford_basic and uncheck `Use the logo supplied by the theme` so you can upload an image. 
12. Upload a large image.
13. Save the form and review the home page.
14. See only the uploaded logo and at a reasonable size.
15. Go back to /admin/appearance/settings/stanford_basic and select `Option A` from the list of options and fill out the lines.
16. Save the form and go to the home page to see your uploaded logo and the lockup text. 

# Affected Projects or Products
- ACSF Stacks
- SAA/OOD
- SOE
- stanford_profile
- Earth

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/D8CORE-1472
- https://stanfordits.atlassian.net/browse/D8CORE-1768
- https://stanfordits.atlassian.net/browse/D8CORE-1771
- https://github.com/SU-SWS/stanford_profile/pull/220
- @rmundstock 
- @buttonwillowsix 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
Ok, looking at this, it looks like the theme didn't support a custom logo on the 4.x branch so the solution for the custom logo from the config page was to avoid rendering the lockup altogether? Was this discussed with Ishi? I would think there would be a case where we would want the logo to co-exist with the lockup line options.

**Comment by sherakama** [comment]
Conversation on above comment started here: https://stanfordwebservices.slack.com/archives/GJQJ7HTK7/p1594246664428000

File: templates/block/block--system-branding-block.html.twig [hunk 1]
```diff
@@ -14,34 +14,42 @@
   Add that logic here.
 #}
 
-{% if lockup.option is not empty %}
-  {% set attributes = attributes.addClass(["su-lockup", "su-lockup--option-" ~ lockup.option]) %}
+{% if su_use_theme_logo == "0" and su_path_to_custom_logo is not empty %}
+<img src="{{ su_path_to_custom_logo }}" alt="site logo" />
```
Comment (sherakama) [inline templates/block/block--system-branding-block.html.twig:18]: Alt text should be the site title or something significant or nothing at all.

Comment (sherakama) [reply templates/block/block--system-branding-block.html.twig:18]: Logo should link to the front page.

**Comment by sherakama** [comment]
@boznik 

Based on our discussions in slack the first version of this functionality is to support whole lockup replacement but we will need to continue to iterate on this functionality to also support mixed logo and text support. Can you create a ticket and link it here or outline what your development plan is to be able to upgrade or transition into supporting both whole lockup replacement and mixed?

As an aside, it would be preferable that the config page form not introduce functionality that isn't possible in the theme settings as well. We support this theme for systems outside of our product and can add a little of our own custom magic but I would expect that the logo upload functionality that works on the config page also works in the default theme settings.

**Comment by boznik** [comment]
> Ok, looking at this, it looks like the theme didn't support a custom logo on the 4.x branch so the solution for the custom logo from the config page was to avoid rendering the lockup altogether? Was this discussed with Ishi? I would think there would be a case where we would want the logo to co-exist with the lockup line options.

Yes this was discussed with ishi. this version was to account for sites such as:
https://knight-hennessy.stanford.edu/
and
https://neuroscience.stanford.edu/

There was a discussion with ishi and cynthia to add the functionality for a logo with the lockup options in a separate ticket.

**Comment by sherakama** [comment]
@buttonwillowsix @rmundstock 

I'm working to get a preview environment up and will notify you shortly.

**Comment by sherakama** [comment]
@jenbreese You are lead reviewer on this. @imonroe for a second set of eyes.

**Comment by jenbreese** [comment]
@sherakama and @imonroe Ok! I will look.

**Comment by sherakama** [comment]
@katrialesser @rebeccahongsf FYI.

**Comment by jenbreese** [comment]
I tested on Chrome, FF and Safari. The form is easy to use and does what I would expect. I also tried it on different responsive sizes. @imonroe Did you try it out yet?

**Comment by jenbreese** [comment]
@sherakama I tested  again with the updated twig and still looking good.  Do you have more edits? Want me to wait for @imonroe to look?

**Comment by sherakama** [comment]
We can give it another day and then roll it in at the end of the sprint tomorrow.

**Comment by sherakama** [comment]
@jenbreese This is good to merge now.

**Pull request merged**